hpcflow/sdk/helper/helper.py

```diff
@@ -136,6 +139,54 @@ def start_helper(
             sys.exit(1)
 
 
+def modify_helper(
+    app,
+    timeout=DEFAULT_TIMEOUT,
+    timeout_check_interval=DEFAULT_TIMEOUT_CHECK,
+    watch_interval=DEFAULT_WATCH_INTERVAL,
+):
+    PID_file = get_PID_file_path(app)
+    if PID_file.is_file():
+        [helper_pid, to, tci, wi] = strip_helper_PID(app)
+        if (
+            float(to) != timeout
+            or float(tci) != timeout_check_interval
+            or float(wi) != watch_interval
+        ):
+            logger = get_helper_logger(app)
+            logger.info(
+                f"Modifying helper with pid={helper_pid}"
+                f" to: timeout={timeout!r}, timeout_check_interval="
+                f"{timeout_check_interval!r} and watch_interval={watch_interval!r}."
+            )
+
+            if isinstance(timeout, timedelta):
+                timeout = timeout.total_seconds()
+            if isinstance(timeout_check_interval, timedelta):
+                timeout_check_interval = timeout_check_interval.total_seconds()
+            if isinstance(watch_interval, timedelta):
+                watch_interval = watch_interval.total_seconds()
+
+            try:
+                with PID_file.open("wt") as fp:
+                    fp.write(f"pid = {helper_pid}\n")
+                with PID_file.open("at") as fp:
+                    fp.write(f"timeout = {timeout}\n")
+                    fp.write(f"timeout-check-interval = {timeout_check_interval}\n")
+                    fp.write(f"watch-interval = {watch_interval}\n")
+            except FileNotFoundError as err:
+                logger.error(
+                    f"Could not write to the PID file {PID_file!r}; killing helper process. "
+                    f"Exception was: {err!r}"
+                )
+                proc.kill()
```

Comment:
`proc` is not defined; remove this and remove the "killing helper process" from the error message.

Reply:
Resolved in ac861f7